File: hardToTest/GcdLcm/GcdLcm.java
public class GcdLcm {
    public static void gcdLcm(int a, int b) {
        if (a <= 0 || b <= 0) {
            System.out.println("GCD: null, LCM: null");
            return;
        }

        int x = a, y = b;
        while (y != 0) {
            int temp = y;
            y = x % y;
            x = temp;
        }
        int gcd = x;
        int lcm = (a * b) / gcd;

        System.out.println("GCD: " + gcd + ", LCM: " + lcm);
    }

    public static void main(String[] args) {
        int a = 15, b = 20;
        gcdLcm(a, b);
    }
}

File: hardToTest/GcdLcm/gcd_lcm.py
def gcd_lcm(a, b):
    if a <= 0 or b <= 0:
        return None, None
    
    # 计算 GCD 和 LCM
    x, y = a, b
    while y:
        x, y = y, x % y
    gcd = x
    lcm = (a * b) // gcd
    return gcd, lcm

# 示例
a = 15
b = 20
gcd, lcm = gcd_lcm(a, b)
print(f"GCD: {gcd}, LCM: {lcm}")